pkg/envoy/route_test.go

```diff
@@ -0,0 +1,57 @@
+package envoy
+
+import (
+	"reflect"
+	"testing"
+
+	envoy_route_v3 "github.com/envoyproxy/go-control-plane/envoy/config/route/v3"
+	networkingv1 "k8s.io/api/networking/v1"
+)
+
+func notIsType(a, b interface{}) bool {
+	return reflect.TypeOf(a) != reflect.TypeOf(b)
+}
+
+func TestPrefixStringPath(t *testing.T) {
+	ingress := newIngress("foo.app.com", "foo.cluster.com", pathvars{"path": "/", "pathType": networkingv1.PathTypePrefix})
+	c := translateIngresses([]networkingv1.Ingress{ingress})
+
+	pathtype := c.VirtualHosts[0].Routes[0].Route.GetPathSpecifier()
+
+	if notIsType(pathtype, &envoy_route_v3.RouteMatch_Prefix{}) {
+		t.Errorf("expected pathType for foo.app.com ingress , was %s", reflect.TypeOf(pathtype))
+	}
+}
+
+func TestPrefixPath(t *testing.T) {
+	ingress := newIngress("foo.app.com", "foo.cluster.com", pathvars{"path": "/testing", "pathType": networkingv1.PathTypePrefix})
+	c := translateIngresses([]networkingv1.Ingress{ingress})
+
+	pathtype := c.VirtualHosts[0].Routes[0].Route.GetPathSpecifier()
+
+	if notIsType(pathtype, &envoy_route_v3.RouteMatch_PathSeparatedPrefix{}) {
```

Comment:
Should we test here that the path actually translates to /testing ?

Reply:
we are checking if the translate function works properly and give us correct type for route

Reply:
yes, but I would like to see us test tht the route is actually correct, not only right type

Reply:
I updated the code, please review it again 
I added path testing too